tests/test_coil.py

```diff
@@ -525,3 +525,44 @@ def test_word_swap_unset(size, encoder: CoilDataEncoder):
 
     with pytest.raises(EncodeException):
         encoder.encode(CoilData(coil, 1))
+
+
+# Date
+@pytest.fixture
+def coil_date():
+    return Coil(
+        123,
+        "holiday-48044",
+        "Holiday",
+        "u16",
+        type="date",
+    )
+
+
+@pytest.mark.parametrize(
+    "raw_value, value",
+    [
+        (b"\x01\x00", datetime.date(2007, 1, 2)),
+        (b"\xbf\x17", datetime.date(2023, 8, 24)),
+        (b"\xc5\x17", datetime.date(2023, 8, 30)),
+    ],
+)
+def test_date_decode(
+    raw_value, value, encoder_word_swap: CoilDataEncoder, coil_date: Coil
+):
+    assert encoder_word_swap.decode(coil_date, raw_value) == CoilData(coil_date, value)
+
+
+@pytest.mark.parametrize(
+    "value, raw_value",
+    [
+        (datetime.date(2007, 1, 2), b"\x01\x00\x00\x00"),
+        (datetime.date(2023, 8, 24), b"\xbf\x17\x00\x00"),
+        (datetime.date(2023, 8, 30), b"\xc5\x17\x00\x00"),
```

Comment:
We should add a date here with timezone attached. So we at least keep track of how it behaves compared to a naive date time.

Reply:
`datetime.date` does not have a time zone.